### ISSUE TITLE
documenting purpose of webcompat.com labels

### ISSUE THREAD

**Issue by zoepage** [webcompat/webcompat.com#1483]
Atm we have 32 labels: 

```
7 open issues in progress
6 open issues lang: CSS
1 open issue lang: HTML
15 open issues lang: JS
0 open issues lang: Markdown
9 open issues lang: Python
9 open issues ready
27 open issues scope: design
15 open issues scope: features
0 open issues scope: outreachy
1 open issue scope: security
10 open issues scope: testing
27 open issues site-development
5 open issues status: accepted
0 open issues status: add to changelog
3 open issues status: blocked
44 open issues status: discussion
9 open issues status: good first bug
17 open issues status: good next bug
7 open issues status: in progress
1 open issue status: leave open
1 open issue status: needs decision
0 open issues status: needs review
6 open issues status: needs tests
19 open issues status: ready
26 open issues type: bug
7 open issues type: documentation
12 open issues type: feature request
0 open issues type: regression
1 open issues type: backlog
x open issues type: content
x open issues type: refactor
```

Some of them are pretty straight forward like the `needs xyz`, `type: a` or `lang: b` labels.
Some are clear from the meaning, but the process behind it might not, e.g. when do I use `ready` or `in progress` or `status: accepted` and what is the difference between e.g. `ready` and `status: ready`. 

I'd like to document it somewhere and make this more transparent to new contributors. (Probably CONTRIBUTE.md could be a good place for that?)

**Comment by karlcow** [comment]
@zoepage good idea. Some of them are documented in #1097

**Comment by magsout** [comment]
@zoepage good idea.



I would say that `ready` and `in progress` were created by waffle.

**Comment by karlcow** [comment]
yes waffle injection.

**Comment by zoepage** [comment]
This list is a WIP.

## general
Bot related labels.

#### ready
#### site-development
#### in progress


## lang
Language relevant to issue
Main purpose is helping contributors with filtering the issues.

#### lang: CSS
CSS is relevant to that issue.

#### lang: HTML
HTML is relevant to that issue.

#### lang: JS
Javascript is relevant to that issue.

#### lang: Markdown
Markdown is relevant to that issue.

#### lang: Python
Python is relevant to that issue.


## scope
Provides context about the scope in the issue.
Main purpose is helping contributors with filtering the issues.

#### scope: design
Issues related to design / UX and design discussions.

#### scope: features
Issues related to features and feature requests.

#### scope: outreachy
Issues related to the [outreachy](https://www.gnome.org/outreachy/) program.

#### scope: security
Issues related to security.

#### scope: testing
Issues related to testing, setup of tests, writing and running tests.


## status
Indicates the status of a bug.
Main purpose is helping contributors with identify the progress of the issues.

#### status: accepted
The proposal in the issue got accepted and is free to progress.

#### status: add to changelog
For merged pull requests, so we can add the change to the [CHANGELOG.md|(https://github.com/webcompat/webcompat.com/blob/master/CHANGELOG.md).

#### status: blocked
The issue is blocked by another issue (referenced in a comment of the issue) or e.g. missing information (also specified in the comments).

#### status: discussion
The discussion for this issue is open and everyone can provide feedback.

#### status: good first bug
Great first issue to start with the project. Usually the scope of the issue is small, so the contributor can focus on the project set up and getting familiar with workflows etc.

#### status: good next bug
Great issue someone more familiar with the project can grab and work on it. Scope is usually between mediate or bigger.

#### status: in progress

#### status: leave open (could go due the newly created label backlog?)

#### status: needs decision

#### status: needs review (still needed?)

#### status: needs tests

#### status: ready (different to accepted? could be just ditch it?)



## type

#### type: bug
#### type: documentation
#### type: feature request
#### type: regression
#### type: backlog
#### type: content
#### type: refactor

**Comment by zoepage** [comment]
@karlcow @magsout @adamopenweb @miketaylr (and who ever feels like adding input):
Could you please check, if the description of the labels are correct and if there is anything we could delete / change? Would be really appreciated. <3

Thank you.

**Comment by karlcow** [comment]
Thanks @zoepage for doing this. Very useful.

**Comment by magsout** [comment]
👍 very good.

Why do we keep the labels of the bots?

**Comment by miketaylr** [comment]
> Why do we keep the labels of the bots?

Good question. Should we ditch waffle and move to GitHub projects, or just milestones? I log in once a month, if that. But happy to let it exist if at least 1 person finds value.

**Comment by zoepage** [comment]
+1 on GH projects. 

> But happy to let it exist if at least 1 person finds value.

Also +1 on that.

**Comment by miketaylr** [comment]
These look good, thanks @zoepage!

**Comment by karlcow** [comment]
## AGAINST
Here there is one comparison in between waffle and github projects.
https://what.thedailywtf.com/topic/22435/github-projects-vs-waffle
from the post and the comments (if you can pass the obnoxious ones). GH projects seem to require more work from users to manage issues.

Also 
* https://news.ycombinator.com/item?id=12519912

## FOR
https://github.com/Lamassau/PiTrol/issues/26 Here I like the comment saying, let's try for a couple of days.

My usual thought is:

1. Are there things we are not satisfied with in waffle? (my answer: no)
2. Are there features we think would help us in GitHub Projects? (my answer: I don't know)

My natural bias would say I'm fine with waffle. But if a compelling argument is made, why not? Just less not spend 10 hours on this ;)

**Comment by karlcow** [comment]
Also it seems we are using Milestones as Projects? Maybe
https://help.github.com/articles/about-project-boards/

>  Project boards on GitHub help you organize and prioritize your work. You can create project boards for specific feature work, comprehensive roadmaps, or even release checklists. 

Something to take into consideration

> Anyone with read access to a repository can view the repository's project boards. To create a project board, you must have write access to the repository.

**Comment by zoepage** [comment]
> Just less not spend 10 hours on this ;)

+1 !

> Are there things we are not satisfied with in waffle? (my answer: no)

Thank you @karlcow for the information.
Pro for GH Projects for me would be, not to have an additional tool. 

As GH Projects seem like something in an early stage, we could also just wait with the move. :)
Maybe a clearer documentation about our workflow and the waffle.io board including a prominent link somewhere could also help to integrate the habit of using it for collaborators?

I got wondering, so when (most?) of us do not use it, what was the main purpose of adding that board in general? Was it before we started to work with Milestones? Would it make more sense to ditch waffle.io / GH projects in general and work with Milestones for now? What would be the Pros/Cons for (just) Milestones?

**Comment by karlcow** [comment]
@calexity proposed in #644 that we use a project dashboard for the issues. After searching for a tool I proposed waffle which has a tight integration with Gihub.

**Comment by zoepage** [comment]
@karlcow ok. thanks for that context.

As we (mostly) do not use it, do we need it?

**Comment by karlcow** [comment]
I'm using it time to time. Like today for example :)

**Comment by karlcow** [comment]
Tied to this dicussion
https://groups.google.com/forum/#!topic/mozilla.compatibility/wtr-ZuGcKxk

**Comment by karlcow** [comment]
@zoepage what would be the resolution for this issue. Any action item that would satisfy the issue.

You said at the beginning 
> I'd like to document it somewhere and make this more transparent to new contributors. (Probably CONTRIBUTE.md could be a good place for that?)

**Comment by zoepage** [comment]
@karlcow 
> what would be the resolution for this issue. Any action item that would satisfy the issue.

This is content that needs to be added to the website after the refactor into the FAQ or contribute section. But this needs to be build after we've finished the refactor.

So a good first step would be to add this to either the wiki and link to it in our docs or add this to our docs itself, e.g. here https://github.com/webcompat/webcompat.com/blob/master/CONTRIBUTING.md#filing-an-issue

**Comment by zoepage** [comment]
As this is mostly outdated, I'll close.

**Comment by karlcow** [comment]
paper trail. https://groups.google.com/forum/#!topic/mozilla.compatibility/Y9fceIecTME
